### PR TITLE
feat(KSegmentedControl): switch to setup syntax [KHCP-6514]

### PR DESCRIPTION
**Pull request by DariaYeremina**
# Summary

https://konghq.atlassian.net/browse/KHCP-6514
Switch to setup syntax

## PR Checklist

* [ ] Does not introduce dependencies
* [ ] **Functional:** all changes do not break existing APIs and if so, bump major version.
* [ ] **Tests pass:** check the output of yarn test
* [ ] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
* [ ] **Framework style:** abides by the essential rules in Vue's style guide
* [ ] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs, debugger), or leftover comments
* [ ] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

File: src/components/KSegmentedControl/KSegmentedControl.vue [hunk 1]
```diff
@@ -24,102 +24,52 @@
   </div>
 </template>
 
-<script lang="ts">
-import { defineComponent, ref, PropType } from 'vue'
+<script lang="ts" setup>
+import { ref, PropType } from 'vue'
 import KButton from '@/components/KButton/KButton.vue'
+import { SegmentedControlOption } from '@/types/segmented-control'
+// need to be imported as const from separate file due to an error "defineProps are referencing locally declared variables"
```
Comment (kaiarrowood) [inline src/components/KSegmentedControl/KSegmentedControl.vue:31]: You can fix this by including the items in a separate `<script lang="ts">` block (no `setup`) like we are doing here: https://github.com/Kong/kongponents/blob/main/src/components/KMultiselect/KMultiselect.vue#L244

If you have the non-`setup` `script` block BEFORE the `setup` block you need to move all `import` statements into the non-`setup` block. If you have the non-`setup` block AFTER the `setup` block you only have to move the functions being used in `defineProps`.

Comment (DariaYeremina) [reply src/components/KSegmentedControl/KSegmentedControl.vue:31]: then I got another error with duplicated import of type

Comment (DariaYeremina) [reply src/components/KSegmentedControl/KSegmentedControl.vue]: updated

File: src/components/KSegmentedControl/KSegmentedControl.vue [hunk 2]
```diff
@@ -71,55 +66,50 @@ const validateItems = (items: SegmentedControlOption[] | string[]): boolean => {
...
-    },
-    isDisabled: {
-      type: Boolean,
-      default: false,
-    },
-    allowPointerEvents: {
-      type: Boolean,
-      default: false,
-    },
+export default {}
```
Comment (kaiarrowood) [inline src/components/KSegmentedControl/KSegmentedControl.vue:69]: nit: I think you can remove this line

**Review by kaiarrowood — APPROVED** [review]
Looking good 🚀

**Pull request merged**